dotnet publish -p ContainerImageTags='"1.2.3-alpha2;latest"' throws "MSB1006: Invalid property"

### Type of issue

Code doesn't work

### Description

From https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-7-0, there is a code snippet, stating that docker build image tags can be set with e.g. `dotnet publish -p ContainerImageTags='"1.2.3-alpha2;latest"'`. However, this doesn't work and throws a MSB1006 error:

```log
MSBUILD : error MSB1006: Die Eigenschaft ist ungültig.
Schalter: latest'
```

MSBuild-Version 17.8.3+195e7f5a3 for .NET

Hint:
Specifying one tag works, specifying more than one tag gives the error.

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-7-0

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/docker/publish-as-container.md

### Document Version Independent Id

401bc84b-9410-3583-91f6-88968654b588

### Article author

@IEvangelist

### Metadata

* ID: 19c28091-b3ad-affb-6343-47d7fa066d55 
* Product: **dotnet-fundamentals**
* Technology: **dotnet-docker**

--- a/docs/core/docker/publish-as-container.md
+++ b/docs/core/docker/publish-as-container.md
@@ -386,6 +386,13 @@ Tags can only contain up to 127 alphanumeric characters, periods, underscores, a
 >
 > This results in two images being generated: `my-app:1.2.3-alpha2` and `my-app:latest`.
 
+> [!TIP]
+> If you experience issues with the `ContainerImageTags` property, consider scoping an environment variable `ContainerImageTags` instead:
+>
+> ```dotnetcli
+> ContainerImageTags='1.2.3;latest' dotnet publish
+> ```
+
 ### `ContainerLabel`
 
 The container label adds a metadata label to the container. Labels have no impact on the container at run time, but are often used to store version and authoring metadata for use by security scanners and other infrastructure tools. You can specify any number of container labels.

--- a/docs/core/docker/publish-as-container.md
+++ b/docs/core/docker/publish-as-container.md
@@ -1,7 +1,7 @@
 ---
 title: Containerize an app with dotnet publish
 description: In this tutorial, you'll learn how to containerize a .NET application with dotnet publish.
-ms.date: 12/18/2023
+ms.date: 01/10/2024
 ms.topic: tutorial
 zone_pivot_groups: dotnet-version-7-8
 ---
@@ -182,6 +182,46 @@ You can control many aspects of the generated container through MSBuild properti
 > [!NOTE]
 > The only exceptions to this are `RUN` commands. Due to the way containers are built, those cannot be emulated. If you need this functionality, you'll need to use a _Dockerfile_ to build your container images.
 
+### Container image naming configuration
+
+Container images follow a specific naming convention. The name of the image is composed of several parts, the registry, optional port, repository, and optional tag and family.
+
+```dockerfile
+REGISTRY[:PORT]/REPOSITORY[:TAG(S)[-FAMILY]]
+```
+
+For example, `mcr.microsoft.com/dotnet/runtime:8.0-alpine`.
+
+Some properties described in the following sections correspond to managing parts of the image name. Consider the following table that maps the relationship between the image name and the properties:
+
+:::zone pivot="dotnet-8-0"
+
+| Image name part   | MSBuild property      | Example                                       |
+|-------------------|-----------------------|-----------------------------------------------|
+| `REGISTRY[:PORT]` | `ContainerRegistry`   | `mcr.microsoft.com:443`                       |
+| `PORT`            | `ContainerPort`       | `:443`                                        |
+| `REPOSITORY`      | `ContainerRepository` | `dotnet/runtime`                              |
+| `REPOSITORY`      | `ContainerImageName` | `dotnet/runtime`                              |
+| `TAG`             | `ContainerImageTag`   | `8.0`                                         |
+| `TAGS`            | `ContainerImageTags`  | `8.0;latest`                                  |
+| `FAMILY`          | `ContainerFamily`     | `-alpine`                                     |
+| `*`               | `ContainerImageName`  | `mcr.microsoft.com/dotnet/runtime:8.0-alpine` |
+
+:::zone-end
+:::zone pivot="dotnet-7-0"
+
+| Image name part   | MSBuild property      | Example                                       |
+|-------------------|-----------------------|-----------------------------------------------|
+| `REGISTRY[:PORT]` | `ContainerRegistry`   | `mcr.microsoft.com:443`                       |
+| `PORT`            | `ContainerPort`       | `:443`                                        |
+| `REPOSITORY`      | `ContainerImageName`  | `dotnet/runtime`                              |
+| `TAG`             | `ContainerImageTag`   | `8.0`                                         |
+| `TAGS`            | `ContainerImageTags`  | `8.0;latest`                                  |
+| `FAMILY`          | `ContainerFamily`     | `-alpine`                                     |
+| `*`               | `ContainerImageName`  | `mcr.microsoft.com/dotnet/runtime:8.0-alpine` |
+
+:::zone-end
+
 ### `ContainerBaseImage`
 
 The container base image property controls the image used as the basis for your image. By default, the following values are inferred based on the properties of your project:
@@ -215,7 +255,7 @@ If you set a value here, you should set the fully qualified name of the image to
 
 :::zone pivot="dotnet-8-0"
 
-### ContainerFamily
+### `ContainerFamily`
 
 Starting with .NET 8, you can use the `ContainerFamily` MSBuild property to choose a different family of Microsoft-provided container images as the base image for your app. When set, this value is appended to the end of the selected TFM-specific tag, changing the tag provided. For example, to use the Alpine Linux variants of the .NET base images, you can set `ContainerFamily` to `alpine`:
 
@@ -247,7 +287,7 @@ For more information regarding the runtime identifiers supported by .NET, see [R
 
 ### `ContainerRegistry`
 
-The container registry property controls the destination registry, the place that the newly created image will be pushed to. By default it's pushed to the local Docker daemon, but you can also specify a remote registry.  When using a remote registry that requires authentication, you authenticate using the well-known `docker login` mechanisms. For more information, See [authenticating to container registries](https://aka.ms/dotnet/containers/auth) for more details. For a concrete example of using this property, consider the following XML example:
+The container registry property controls the destination registry, the place that the newly created image will be pushed to. By default it's pushed to the local Docker daemon, but you can also specify a remote registry. When using a remote registry that requires authentication, you authenticate using the well-known `docker login` mechanisms. For more information, See [authenticating to container registries](https://aka.ms/dotnet/containers/auth) for more details. For a concrete example of using this property, consider the following XML example:
 
 ```xml
 <PropertyGroup>
@@ -255,7 +295,7 @@ The container registry property controls the destination registry, the place tha
 </PropertyGroup>
 ```
 
-This tooling supports publishing to any registry that supports the [Docker Registry HTTP API V2](https://docs.docker.com/registry/spec/api/).  This includes the following registries explicitly (and likely many more implicitly):
+This tooling supports publishing to any registry that supports the [Docker Registry HTTP API V2](https://docs.docker.com/registry/spec/api/). This includes the following registries explicitly (and likely many more implicitly):
 
 * [Azure Container Registry](https://azure.microsoft.com/products/container-registry)
 * [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/)

--- a/docs/core/docker/publish-as-container.md
+++ b/docs/core/docker/publish-as-container.md
@@ -187,7 +187,7 @@ You can control many aspects of the generated container through MSBuild properti
 Container images follow a specific naming convention. The name of the image is composed of several parts, the registry, optional port, repository, and optional tag and family.
 
 ```dockerfile
-REGISTRY[:PORT]/REPOSITORY[:TAG(S)[-FAMILY]]
+REGISTRY[:PORT]/REPOSITORY[:TAG[-FAMILY]]
 ```
 
 For example, consider the fully qualified `mcr.microsoft.com/dotnet/runtime:8.0-alpine` image name:
@@ -206,7 +206,6 @@ Some properties described in the following sections correspond to managing parts
 | `PORT`            | `ContainerPort`       | `:443`                  |
 | `REPOSITORY`      | `ContainerRepository` | `dotnet/runtime`        |
 | `TAG`             | `ContainerImageTag`   | `8.0`                   |
-| `TAGS`            | `ContainerImageTags`  | `8.0;latest`            |
 | `FAMILY`          | `ContainerFamily`     | `-alpine`               |
 
 :::zone-end
@@ -218,10 +217,10 @@ Some properties described in the following sections correspond to managing parts
 | `PORT`            | `ContainerPort`      | `:443`                  |
 | `REPOSITORY`      | `ContainerImageName` | `dotnet/runtime`        |
 | `TAG`             | `ContainerImageTag`  | `8.0`                   |
-| `TAGS`            | `ContainerImageTags` | `8.0;latest`            |
-| `FAMILY`          | `ContainerFamily`    | `-alpine`               |
 
 :::zone-end
+
+The following sections describe the various properties that can be used to control the generated container image.
 
 ### `ContainerBaseImage`
 

--- a/docs/core/docker/publish-as-container.md
+++ b/docs/core/docker/publish-as-container.md
@@ -196,29 +196,26 @@ Some properties described in the following sections correspond to managing parts
 
 :::zone pivot="dotnet-8-0"
 
-| Image name part   | MSBuild property      | Example                                       |
-|-------------------|-----------------------|-----------------------------------------------|
-| `REGISTRY[:PORT]` | `ContainerRegistry`   | `mcr.microsoft.com:443`                       |
-| `PORT`            | `ContainerPort`       | `:443`                                        |
-| `REPOSITORY`      | `ContainerRepository` | `dotnet/runtime`                              |
-| `REPOSITORY`      | `ContainerImageName` | `dotnet/runtime`                              |
-| `TAG`             | `ContainerImageTag`   | `8.0`                                         |
-| `TAGS`            | `ContainerImageTags`  | `8.0;latest`                                  |
-| `FAMILY`          | `ContainerFamily`     | `-alpine`                                     |
-| `*`               | `ContainerImageName`  | `mcr.microsoft.com/dotnet/runtime:8.0-alpine` |
+| Image name part   | MSBuild property      | Example values          |
+|-------------------|-----------------------|-------------------------|
+| `REGISTRY[:PORT]` | `ContainerRegistry`   | `mcr.microsoft.com:443` |
+| `PORT`            | `ContainerPort`       | `:443`                  |
+| `REPOSITORY`      | `ContainerRepository` | `dotnet/runtime`        |
+| `TAG`             | `ContainerImageTag`   | `8.0`                   |
+| `TAGS`            | `ContainerImageTags`  | `8.0;latest`            |
+| `FAMILY`          | `ContainerFamily`     | `-alpine`               |
 
 :::zone-end
 :::zone pivot="dotnet-7-0"
 
-| Image name part   | MSBuild property      | Example                                       |
-|-------------------|-----------------------|-----------------------------------------------|
-| `REGISTRY[:PORT]` | `ContainerRegistry`   | `mcr.microsoft.com:443`                       |
-| `PORT`            | `ContainerPort`       | `:443`                                        |
-| `REPOSITORY`      | `ContainerImageName`  | `dotnet/runtime`                              |
-| `TAG`             | `ContainerImageTag`   | `8.0`                                         |
-| `TAGS`            | `ContainerImageTags`  | `8.0;latest`                                  |
-| `FAMILY`          | `ContainerFamily`     | `-alpine`                                     |
-| `*`               | `ContainerImageName`  | `mcr.microsoft.com/dotnet/runtime:8.0-alpine` |
+| Image name part   | MSBuild property     | Example values          |
+|-------------------|----------------------|-------------------------|
+| `REGISTRY[:PORT]` | `ContainerRegistry`  | `mcr.microsoft.com:443` |
+| `PORT`            | `ContainerPort`      | `:443`                  |
+| `REPOSITORY`      | `ContainerImageName` | `dotnet/runtime`        |
+| `TAG`             | `ContainerImageTag`  | `8.0`                   |
+| `TAGS`            | `ContainerImageTags` | `8.0;latest`            |
+| `FAMILY`          | `ContainerFamily`    | `-alpine`               |
 
 :::zone-end
 
@@ -341,7 +338,12 @@ Image names consist of one or more slash-delimited segments, each of which can o
 
 ### `ContainerImageTag(s)`
 
-The container image tag property controls the tags that are generated for the image. To specify a single tag use `ContainerImageTag` and for multiple tags use `ContainerImageTags`. Tags are often used to refer to different versions of an app, but they can also refer to different operating system distributions, or even different configurations.
+The container image tag property controls the tags that are generated for the image. To specify a single tag use `ContainerImageTag` and for multiple tags use `ContainerImageTags`.
+
+> [!IMPORTANT]
+> When you use `ContainerImageTags`, you'll end up with multiple images, one per unique tag.
+
+Tags are often used to refer to different versions of an app, but they can also refer to different operating system distributions, or even different configurations.
 
 :::zone pivot="dotnet-8-0"
 
@@ -378,6 +380,8 @@ Tags can only contain up to 127 alphanumeric characters, periods, underscores, a
 > ```dotnetcli
 > dotnet publish -p ContainerImageTags='"1.2.3-alpha2;latest"'
 > ```
+>
+> This results in two images being generated: `my-app:1.2.3-alpha2` and `my-app:latest`.
 
 ### `ContainerLabel`
 

--- a/docs/core/docker/publish-as-container.md
+++ b/docs/core/docker/publish-as-container.md
@@ -190,9 +190,13 @@ Container images follow a specific naming convention. The name of the image is c
 REGISTRY[:PORT]/REPOSITORY[:TAG(S)[-FAMILY]]
 ```
 
-For example, `mcr.microsoft.com/dotnet/runtime:8.0-alpine`.
+For example, consider the fully qualified `mcr.microsoft.com/dotnet/runtime:8.0-alpine` image name:
 
-Some properties described in the following sections correspond to managing parts of the image name. Consider the following table that maps the relationship between the image name and the properties:
+- `mcr.microsoft.com` is the registry (and in this case represents the Microsoft container registry).
+- `dotnet/runtime` is the repository (but some consider this the `user/repository`).
+- `8.0-alpine` is the tag and family (the family is an optional specifier that helps disambiguate OS packaging).
+
+Some properties described in the following sections correspond to managing parts of the generated image name. Consider the following table that maps the relationship between the image name and the build properties:
 
 :::zone pivot="dotnet-8-0"
 

